### PR TITLE
Add ContainerPushRepository type

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #6825
https://pulp.plan.io/issues/6825

### CODE REVIEW

File: pulp_container/app/viewsets.py [hunk 1]
```diff
@@ -560,6 +564,63 @@ def __init__(self, blob, path, status, request, send_body=False):
...
+        dist_created = False
+        if create:
+            # TODO: Use serializer to create objects
+            distribution, dist_created = models.ContainerDistribution.objects.get_or_create(
+                base_path=path, defaults={"name": path},
+            )
+        else:
+            distribution = get_object_or_404(models.ContainerDistribution, base_path=path)
+        if dist_created:
+            repository = models.ContainerPushRepository.objects.create(name=path)
```
Comment (ipanova) [inline pulp_container/app/viewsets.py:608]: can we add here try/except in case there is already a ContainerRepository with such name?

Comment (ipanova) [reply pulp_container/app/viewsets.py:608]: also maybe we should delete the distribution if we fail with repo creation

Comment (mdellweg) [reply pulp_container/app/viewsets.py:608]: Sure. What should be in the except block? Is it a 404, 409 or 500?
What if there is already a distribution, but no repository. Is auto creating the latter ok? Or does it mean someone reserved that name for later use?

Comment (mdellweg) [reply pulp_container/app/viewsets.py:608]: We should put repo and distribution creation into a database transaction.

Comment (ipanova) [reply pulp_container/app/viewsets.py:608]: and repo_version

Comment (ipanova) [reply pulp_container/app/viewsets.py:608]: as we discussed per call - if existing distribution has no repo/repo_version assigned to it --> fail whole transaction

**Comment by ipanova** [comment]
@mdellweg looks good to me, thank you for this work

**Pull request merged**